src/_P077_CSE7766.ino

```diff
@@ -165,40 +164,26 @@ boolean Plugin_077(byte function, struct EventStruct *event, String& string)
     case PLUGIN_SERIAL_IN:
       {
         if (Plugin_077_init) {
-          if (Serial.available() > 0) {
-            serial_in_byte = Serial.read();
-            success = true;
-
-            if (cse_receive_flag) {
-              serial_in_buffer[serial_in_byte_counter++] = serial_in_byte;
-              if (24 == serial_in_byte_counter) {
-                addLog(LOG_LEVEL_DEBUG_DEV, F("CSE: packet received"));
-                CseReceived(event);
-                cse_receive_flag = 0;
-                serial_in_byte_counter = 0;
-                // new packet received, update values
-                UserVar[event->BaseVarIndex] = energy_voltage;
-                UserVar[event->BaseVarIndex + 1] = energy_power;
-                UserVar[event->BaseVarIndex + 2] = energy_current;
-                UserVar[event->BaseVarIndex + 3] = cf_pulses;
-//                break;
-              }
-            } else {
-              if (0x5A == serial_in_byte) { // 0x5A - Packet header 2
-                cse_receive_flag = 1;
-                addLog(LOG_LEVEL_DEBUG_DEV, F("CSE: Header received"));
-              } else {
-                serial_in_byte_counter = 0;
-              }
-              serial_in_buffer[serial_in_byte_counter++] = serial_in_byte;
+          success = true;
+          /* ONLINE CHECKSUMMING by Bartlomiej Zimon*/
+          bool P077_found = false;
+          while (Serial.available()>0 && !P077_found){ // if we have data try to find good checksum
+            uint8_t serial_in_byte = Serial.read();
```

Comment:
How blocking is this while loop + reading?

Reply:
we read as much as packet size if checksum is ok, if not we try to find another packet if data are available.

Reply:
looks like 1ms max

Reply:
have used here memmove as it could be optimized for ESPs